homeassistant/helpers/intent.py

```diff
@@ -252,13 +265,47 @@ def __init__(
         self.text_input = text_input
         self.context = context
         self.language = language
+        self.category = category
 
     @callback
     def create_response(self) -> IntentResponse:
         """Create a response."""
         return IntentResponse(self, language=self.language)
 
 
+class IntentResponseState(Enum):
+    """State of the intent response."""
+
+    SUCCESS = "success"
+    FAILURE = "failure"
+
+
+class IntentResponseType(Enum):
+    """Type of the intent response."""
+
+    ACTION_CONFIRMATION = "action_confirmation"
+    """speech is a summary of action taken"""
+
+    QUERY_ANSWER = "query_answer"
+    """speech is an answer to a question"""
+
+    ERROR_MESSAGE = "error_message"
```

Comment:
this should just be ERROR, it's more than a message.

Reply:
Do you think the error message should be moved out of "speech" and into an "error" dict instead?

Reply:
I think that we can keep speech if that's how we want the user to receive the error message.